lua/effects/eff_term_goodarc.lua

```diff
@@ -0,0 +1,436 @@
+--   Cyan:             Vector( 0.1, 1, 1 )
+--   White:            Vector( 1, 1, 1 )
+--   Purple:           Vector( 0.6, 0.1, 1 )
+--   Pink:             Vector( 1, 0.4, 0.7 )
+--   Orange:           Vector( 1, 0.5, 0.1 )
+-- ============================================
+
+local BeamMaterial = CreateMaterial( "xeno/beamlightning", "UnlitGeneric", {
+    ["$basetexture"] = "sprites/spotlight",
+    ["$additive"] = "1",
+    ["$vertexcolor"] = "1",
+    ["$vertexalpha"] = "1",
+} )
+
+local FLAG_NO_LIGHT = 1
+local FLAG_BRANCHES = 2
+local FLAG_NO_FADE = 4
+local FLAG_CONNECT = 8
+local FLAG_THICK_CORE = 16
+local FLAG_NO_FLICKER = 32
+local FLAG_NO_SOUND = 64
+
+-- Electric sound pool
+local ElectricSounds = {
+    "ambient/energy/zap1.wav",
+    "ambient/energy/zap2.wav",
+    "ambient/energy/zap3.wav",
+    "ambient/energy/zap5.wav",
+    "ambient/energy/zap6.wav",
+    "ambient/energy/zap7.wav",
+    "ambient/energy/zap8.wav",
+    "ambient/energy/zap9.wav",
+}
+
+local SparkSounds = {
+    "ambient/energy/spark1.wav",
+    "ambient/energy/spark2.wav",
+    "ambient/energy/spark3.wav",
+    "ambient/energy/spark4.wav",
+    "ambient/energy/spark5.wav",
+    "ambient/energy/spark6.wav",
+}
+
+function EFFECT:Init( data )
+    self.StartPos = data:GetStart()
+    self.EndPos = data:GetOrigin()
+    self.Scale = math.max( data:GetScale() or 1, 0.1 )
+
+    -- Width multiplier
+    local widthMul = data:GetSurfaceProp()
+    self.WidthMul = ( widthMul > 0 ) and widthMul or 1
+
+    -- Duration multiplier
+    local durMul = data:GetHitBox()
+    self.DurationMul = ( durMul > 0 ) and durMul or 1
+
+    -- Flicker speed multiplier
+    local flickerMul = data:GetMaterialIndex()
+    self.FlickerMul = ( flickerMul > 0 ) and flickerMul or 1
+
+    -- Light brightness multiplier
+    local lightMul = data:GetColor()
+    self.LightMul = ( lightMul > 0 ) and lightMul or 1
+
+    -- Sound settings
+    local soundVol = data:GetAttachment()
+    self.SoundVolume = ( soundVol >= 0 ) and math.Clamp( soundVol, 0, 2 ) or 1
+
+    local soundAng = data:GetAngles()
+    self.SoundPitch = ( soundAng and soundAng.p > 0 ) and math.Clamp( soundAng.p, 50, 200 ) or 100
+
+    -- Timing
+    self.Duration = math.Clamp( 0.06 * self.Scale * self.DurationMul, 0.02, 1 )
+    self.DieTime = CurTime() + self.Duration
+
+    -- Arc segments
+    local segments = data:GetMagnitude()
+    self.Segments = ( segments >= 3 ) and math.Clamp( math.floor( segments ), 3, 64 ) or 8
+
+    -- Jitter intensity
+    local radius = data:GetRadius()
+    self.Jitter = ( radius > 0 ) and radius or ( 20 * self.Scale )
+
+    -- Color from normal vector
+    local col = data:GetNormal()
+    if col:LengthSqr() > 0 then
+        self.Color = Color(
+            math.Clamp( col.x * 255, 0, 255 ),
+            math.Clamp( col.y * 255, 0, 255 ),
+            math.Clamp( col.z * 255, 0, 255 )
+         )
+    else
+        self.Color = Color( 100, 150, 255 )
```

Comment:
You should make a local defaultBeamColor = Color( 100, 150, 255 ) above the EFFECT:Init() code, and just make this use that.
Every Color() call creates a new object, if you instead make it use one color object defined above, less lua garbage to collect.

This would be a MAJOR problem if this was in drawing/rendering code that runs alot.